indexer/processor/l2_processor.go

```diff
@@ -1,18 +1,60 @@
 package processor
 
 import (
+	"context"
+	"errors"
+	"reflect"
+
 	"github.com/ethereum-optimism/optimism/indexer/database"
 	"github.com/ethereum-optimism/optimism/indexer/node"
+	"github.com/google/uuid"
 
+	"github.com/ethereum/go-ethereum"
+	"github.com/ethereum/go-ethereum/common"
 	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/ethclient"
 	"github.com/ethereum/go-ethereum/log"
 )
 
+type L2Contracts struct {
+	L2CrossDomainMessenger common.Address
+	L2StandardBridge       common.Address
+	L2ERC721Bridge         common.Address
+	L2ToL1MessagePasser    common.Address
+
+	// Some more contracts -- ProxyAdmin, SystemConfig, etcc
+	// Ignore the auxiliary contracts?
+
+	// Legacy Contracts? We'll add this in to index the legacy chain.
+	// Remove afterwards?
+}
+
+func L2ContractPredeploys() L2Contracts {
+	return L2Contracts{
+		L2CrossDomainMessenger: common.HexToAddress("0x4200000000000000000000000000000000000007"),
+		L2StandardBridge:       common.HexToAddress("0x4200000000000000000000000000000000000010"),
+		L2ERC721Bridge:         common.HexToAddress("0x4200000000000000000000000000000000000014"),
+		L2ToL1MessagePasser:    common.HexToAddress("0x4200000000000000000000000000000000000016"),
+	}
+}
+
+func (c L2Contracts) toArray() []common.Address {
```

Comment:
nit: this is technically a slice, i usually dont see array language usage in go

Reply:
ah yep good point